src/ast/nodes/AssignmentExpression.ts

```diff
@@ -12,6 +12,8 @@ export default class AssignmentExpression extends NodeBase {
 	bind() {
 		super.bind();
 		this.left.reassignPath([], ExecutionPathOptions.create());
+		// We can not propagate mutations of the new binding to the old binding with certainty
+		this.right.reassignPath([UNKNOWN_KEY], ExecutionPathOptions.create());
```

Comment:
I'm just wondering if there would there be another way to mark these "deoptimizations" that is a little more efficient than instantiating execution path options and the key array?

Reply:
Good idea. As none of these variables are actually mutated, I changed the code to take them from a common pool of initializers. No huge performance gains, though.